tests/unit/orchestration/prefect/test_utils.py

```diff
@@ -0,0 +1,136 @@
+    ]
+    assert all(setup_dates[key] == ddh1.replacements[key] for key in keys_to_compare)
+
+
+def test_process_dates_single(setup_dates):
+    """Test if process_dates function replaces a single date placeholder."""
+    text = "Today's date is <<today>>."
+    replaced_text = ddh1.process_dates(text)
+    expected_text = f"Today's date is {setup_dates['today']}."
+    assert replaced_text == expected_text
+
+
+def test_process_dates_multiple(setup_dates):
+    """Test if process_dates function replaces multiple placeholders."""
+    text = "Today is <<today>>, yesterday was <<yesterday>>"
+    replaced_text = ddh1.process_dates(text)
+    expected_text = (
+        f"Today is {setup_dates['today']}, yesterday was {setup_dates['yesterday']}"
+    )
+    assert replaced_text == expected_text
+
+
+def test_process_dates(setup_dates):
+    """Test if process_dates function replaces multiple placeholders."""
+    text = "The hour and minute are <<now_time>>"
+    replaced_text = ddh1.process_dates(text)
+    expected_text = f"The hour and minute are {setup_dates['now_time'][:-2]}"
+    replaced_text = replaced_text[:-2]
+    assert replaced_text == expected_text
+
+
+def test_process_dates_with_custom_symbols(setup_dates):
+    """Test if process_dates function works with custom start and end symbols."""
+    text = "The year is [[current_year]]."
+    replaced_text = ddh2.process_dates(text)
+    expected_text = f"The year is {setup_dates['today'][:4]}."
+    assert replaced_text == expected_text
+
+
+def test_process_dates_with_malformed_keys():
+    """Test if process_dates function leaves malformed placeholders unchanged."""
+    text = "This is a malformed placeholder: <<today."
+    replaced_text = ddh1.process_dates(text)
+    assert replaced_text == text
+
+
+def test_process_eval_date(setup_dates):
+    """Test if process_dates function works with a pendulum code."""
+    text = "Yesterday was <<(pendulum.today()).subtract(days=1)>>."
+    replaced_text = ddh1.process_dates(text)
+    expected_text = f"Yesterday was {setup_dates['yesterday']}."
+    assert replaced_text == expected_text
+
+
+def test_process_last_x_years():
+    """Test if process_dates function returns a date range of last x years."""
+    x = 5
+    text = f"<<last_{x}_years>>"
+    processed_range = ddh1.process_dates(text)
+    x_years_ago = pendulum.today().year - (x - 1)
+    expected_result = [str(x_years_ago + i) for i in range(x)]
+    assert processed_range == expected_result
+
+
+def test_process_last_x_months():
+    """Test if process_dates function returns a date range of last x years."""
+    x = 18
+    text = f"<<last_{x}_months>>"
+    processed_range = ddh1.process_dates(text)
+    start_date = pendulum.today().subtract(months=(x - 1))
+    expected_result = [
+        start_date.add(months=i).start_of("month").format("YMM") for i in range(x)
+    ]
+    assert processed_range == expected_result
+
+
+def test_Y_years_from_x():
```

Comment:
It's easier to read if you use the same naming convention as in your other tests
```suggestion
def test_process_y_years_from_x():
```